core-test/src/test/scala/org/bitcoins/core/crypto/MnemonicCodeTest.scala

```diff
@@ -0,0 +1,289 @@
+      MnemonicCode.fromEntropy(entropy).words.length == 24
+    }
+
+  }
+
+  it must "not generate back to back idential entropy bitvectors" in {
+    import CryptoGenerators.entropy
+
+    def notEqual(b1: BitVector, b2: BitVector): Assertion = assert(b1 != b2)
+
+    forAll(entropy.bits128, entropy.bits128)(notEqual)
+
+    forAll(entropy.bits160, entropy.bits160)(notEqual)
+
+    forAll(entropy.bits192, entropy.bits192)(notEqual)
+
+    forAll(entropy.bits224, entropy.bits224)(notEqual)
+
+    forAll(entropy.bits256, entropy.bits256)(notEqual)
+
+    forAll(entropy.any, entropy.any)(notEqual)
+  }
+
+  it must "fail to generate mnemonics from entropy of bad length" in {
+    def testEntropyFailure(entropy: BitVector): Assertion = {
+      val short = entropy.drop(1)
+      val long = entropy ++ bin"1"
+      assertThrows[IllegalArgumentException](MnemonicCode.fromEntropy(short))
+      assertThrows[IllegalArgumentException](MnemonicCode.fromEntropy(long))
+    }
+
+    testEntropyFailure(MnemonicCode.getEntropy128Bits)
+    testEntropyFailure(MnemonicCode.getEntropy160Bits)
+    testEntropyFailure(MnemonicCode.getEntropy192Bits)
+    testEntropyFailure(MnemonicCode.getEntropy224Bits)
+    testEntropyFailure(MnemonicCode.getEntropy256Bits)
+  }
+
+  it must "fail to generate mnemonics from too long entropy" in {
+    import CryptoGenerators.entropy
+    forAll(entropy.bits128, entropy.bits160) { (e1, e2) =>
+      val attempt = Try { MnemonicCode.fromEntropy(e1 ++ e2) }
+      assert(attempt.isFailure)
+
+      val exc = attempt
+        .asInstanceOf[Failure[_]]
+        .exception
+
+      val message = exc.getMessage
+
+      message contains "Entropy cannot be longer"
+    }
+  }
+
+  it must "fail to generate mnemonics from too short entropy" in {
+    import CryptoGenerators.entropy
+    forAll(entropy.bits128) { e =>
+      val attempt = Try {
+        MnemonicCode.fromEntropy(
+          // to provide a bitvector of bad length, but correct entropy multiple
+          e.take(MnemonicCode.ENTROPY_MULTIPLE))
+      }
+      assert(attempt.isFailure)
+      val exc = attempt
+        .asInstanceOf[Failure[_]]
+        .exception
+
+      val message = exc.getMessage
+      assert(message contains "Entropy must be at least")
+    }
+  }
+
+  it must "fail to generate mnemonics from entropy of bad length multiple" in {
+    import CryptoGenerators.entropy
+    forAll(entropy.bits192) { e =>
+      val codeT = Try(MnemonicCode.fromEntropy(e.drop(1)))
+
+      assert(codeT.isFailure)
+      val exc = codeT.asInstanceOf[Failure[_]].exception
+      assert(exc.getMessage contains "Entropy must be a multiple")
+    }
+  }
+
+  it must "have serialization symmetry - entropy" in {
+    forAll(CryptoGenerators.mnemonicCode) { code =>
+      val entropy = code.toEntropy
+      val newCode = MnemonicCode.fromEntropy(entropy)
+      assert(code == newCode)
+    }
+  }
+
+  it must "fail on a seed phrase with bad checksum" in {
```

Comment:
I ended up not doing this as a property check. Tried two approaches: 

1) Replace last word in valid phrase
2) Generate a random checksum

The issue was that the checksum is so short that we end up with accidentally producing valid seeds.